website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -16,6 +16,14 @@ import {
 
 let requiredGroupFields = '_id leader tasksOrder name';
 let types = Tasks.tasksTypes.map(type => `${type}s`);
+
+function canNotEditTasks(group, user, assignedUserId) {
+  let isNotGroupLeader = group.leader !== user._id;
+  let isManager = Boolean(group.managers[user._id]);
+  let userIsAssigningToSelf = Boolean(assignedUserId && user._id === assignedUserId);
+  let canNotEditTasks = isNotGroupLeader && !isManager && !userIsAssigningToSelf;
```

Comment:
Yea, that makes sense. The abstraction was completely correct. I'll split the functions.